src/interpreters/wildDice.ts

```diff
@@ -0,0 +1,319 @@
+                
+              case RollStatus.Failure:
+                return `(${rollType}) Disaster${
+                  doubles ? " with a twist" : ""
+                }!`;
+
+                default:
+                throw new UnreachableCaseError(status);
+              }
+            })();
+ 
+      ;}
+
+    })();
+
+    const description =
+       (() => {
+
+          switch (rollType) {
+            // there is probably a way to avoid this deduplication for the twists text
+            case WildType.Action:
+
+              return (() => {
+
+              switch (status) {
+
+                case RollStatus.Full:
+                  return `Complete success, no drawbacks. Mark/clear a box on a track.${
+                    doubles ? " \n\n**(Twist)**\n Adds a small, potentially useful twist, suggested by any player. Firefly has final say." : ""
+                  }`;
+
+                case RollStatus.Mixed:
+                  return `Success with a drawback. Usually marks/ clears a box.${
+                    doubles ? " \n\n**(Twist)**\n Adds a small, potentially useful twist, suggested by any player. Firefly has final say." : ""
+                  }`;
+
+                case RollStatus.Failure:
+                  return `Failure and narrative complication or drawback. Usually doesn't mark/clear a box.${
+                    doubles ? " \n\n**(Twist)**\n Adds a small, potentially useful twist, suggested by any player. Firefly has final say." : ""
+                  }`;
+
+                default:
+                  throw new UnreachableCaseError(status)}
+              })(); 
+              
+            case WildType.Attack:
+              return (() => {
+                switch (status) {
+
+                  case RollStatus.Full:
+                    return `Powerful blow. Deal damage and might inflict an effect.${
+                      doubles ? " \n\n**(Twist... or Critical)**\n Unexpected narrative effect/critical with increased impact." : ""
+                    }`;
+
+                  case RollStatus.Mixed:
+                    return `Attack deals damage and maybe associated effect, but you might take some damage, suffer an effect, lose a resource or be put in a less favourable position.${
+                      doubles ? " \n\n**(Twist... or Critical)**\n Unexpected narrative effect/critical with increased impact." : ""
+                    }`;
+
+                  case RollStatus.Failure:
+                    return `Attack misses or does no damage. You definitely take some damage or an effect, and might lose a resource or be put in a less favourable position too.${
+                      doubles ? " \n\n**(Twist... or Critical)**\n Unexpected narrative effect/critical with increased impact." : ""
+                    }`;
+
+                  default:
+                    throw new UnreachableCaseError(status)}
+                })();
+
+            case WildType.Defense:
+              return (() => {
```

Comment:
these arrow functions are really quite clean given your relative freshness to the concept